Cargo.toml

```diff
@@ -37,6 +37,7 @@ derive_more = { version = "1", default-features = false, features = [
     "from",
     "not",
 ] }
+itertools = { version = "0.13", default-features = false, optional = true }
```

Comment:
we can actually remove this because we only use one function

```
    #[cfg(feature = "use_alloc")]
    fn sorted_unstable_by_key<K, F>(self, f: F) -> VecIntoIter<Self::Item>
    where
        Self: Sized,
        K: Ord,
        F: FnMut(&Self::Item) -> K,
    {
        let mut v = Vec::from_iter(self);
        v.sort_unstable_by_key(f);
        v.into_iter()
    }
```

and we can do that ourselves
then we also don't need to feature gate the state root functions

Reply:
I've removed this and handled it internally, while also cleaning up the feature gate: https://github.com/alloy-rs/trie/pull/65/commits/a29f17f2686d61cd063ea63cf7f4f3e460c665b5